src/client/coll/mod.rs

```diff
@@ -101,24 +160,81 @@ impl<'a> Collection<'a> {
         Ok(cursor.next())
     }
 
+    fn find_one_and_modify(&self, cmd: &mut bson::Document,
+                           filter: bson::Document, max_time_ms: Option<i64>,
+                           projection: Option<bson::Document>, sort: Option<bson::Document>)
+                           -> Result<Option<bson::Document>, String> {
+
+        let mut new_cmd = bson::Document::new();
+        new_cmd.insert("findAndModify".to_owned(), Bson::String(self.name()));
+        new_cmd.insert("query".to_owned(), Bson::Document(filter));
+        if sort.is_some() {
+            new_cmd.insert("sort".to_owned(), Bson::Document(sort.unwrap()));
+        }
+        if projection.is_some() {
+            new_cmd.insert("fields".to_owned(), Bson::Document(projection.unwrap()));
+        }
+
+        for (key, val) in cmd.iter() {
+            new_cmd.insert(key.to_owned(), val.to_owned());
+        }
+
+        let res = try!(self.db.command(new_cmd));
+        match res {
+            Some(doc) => match doc.get("value") {
+                Some(&Bson::Document(ref nested_doc)) => Ok(Some(nested_doc.to_owned())),
+                _ => Ok(None),
+            },
+            None => Ok(None),
+        }
+    }
+
+    fn find_and_update(&self, filter: bson::Document, update: bson::Document,
```

Comment:
I'd recommend renaming this as `find_and_modify` - `find_and_update` is not very mongodb-ish nomenclature. Even if it's private, we should stick to more consistent naming conventions